x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/search_index/connector/connector_configuration_logic.ts

```diff
@@ -86,6 +87,19 @@ function sortConnectorConfiguration(config: ConnectorConfiguration): ConfigEntry
     });
 }
 
+export function ensureStringType(value: string | number | boolean | null): string {
```

Comment:
question: You're now ensuring that these values always match expectations, but that might produce some weird results in cases where configs contain unexpected value types. I'm not sure that forcing them into the right type is the right move, versus actually checking the type and explicitly dealing with type issues. What do you think?

Reply:
I think expanding this is a good idea also. 

> I'm not sure that forcing them into the right type is the right move, versus actually checking the type and explicitly dealing with type issues.

What would you suggest to do to other than converting them to the expected type? Something like if they are the wrong type we don't render the configuration fields and give a detailed error?

Reply:
That sounds good. Do you mind if I do it in a followup PR? I've made a dedicated issue to track it.